### PR TITLE
fix_error_message

### PR DESCRIPTION
**Pull request by shayb**
fix error message when far activitiy not supported

### CODE REVIEW

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt [hunk 1]
```diff
@@ -41,10 +41,10 @@ class TransferActivityModel(private val app: EcosystemApp, var transferActions:
         if (app.transferData != null) {
             val started = transferManager.startTransferRequestActivity(activity, app.identifier, app.transferData.launchActivityFullPath)
             if (!started) {
-                transferActions?.onConnectionError()
+                transferActions?.onStartConnectingError()
             }
         } else {
-            transferActions?.onConnectionError()
+            transferActions?.onStartConnectingError()
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: in this case, best to show the generic error message (this is not supposed to happen)

Comment (shayb) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: ok

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt [hunk 2]
```diff
@@ -64,15 +67,20 @@ class TransferActivityModel(private val app: EcosystemApp, var transferActions:
...
                     }
                 } ?: run {
-                    transferActions?.onConnectionError()
+                    onConnectionError()
                 }
             }
         })
     }
 
+    private fun onConnectionError() {
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: add errorMessage param

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt [hunk 3]
```diff
@@ -41,21 +42,23 @@ class TransferActivityModel(private val app: EcosystemApp, var transferActions:
         if (app.transferData != null) {
             val started = transferManager.startTransferRequestActivity(activity, app.identifier, app.transferData.launchActivityFullPath)
             if (!started) {
-                transferActions?.onConnectionError()
+                transferActions?.onStartConnectingError()
+                analytics.logEvent(Events.Business.CrossAppKinFailure("", Analytics.FAILURE_TYPE_ERROR))
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: error message: "Could not launch "+app.transfer.Data.launchActivityFullPath+" on "+app.identifier

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt [hunk 4]
```diff
@@ -41,21 +42,23 @@ class TransferActivityModel(private val app: EcosystemApp, var transferActions:
...
             }
         } else {
-            transferActions?.onConnectionError()
+            onConnectionError()
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: error message: "Missing launching activity information for "+app.identifier

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt [hunk 5]
```diff
@@ -64,15 +67,20 @@ class TransferActivityModel(private val app: EcosystemApp, var transferActions:
                         userRepository.updateApplicationAddress(app.identifier, it)
                         transferActions?.onConnected()
                     } else {
-                        transferActions?.onConnectionError()
+                        onConnectionError()
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: error message: "Received empty string for public address"

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt [hunk 6]
```diff
@@ -64,15 +67,20 @@ class TransferActivityModel(private val app: EcosystemApp, var transferActions:
...
                     }
                 } ?: run {
-                    transferActions?.onConnectionError()
+                    onConnectionError()
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/TransferActivityModel.kt]: error message: "Received null public address"

**Pull request merged**